zenoh-plugin-ros2dds/src/lib.rs

```diff
@@ -245,27 +300,42 @@ pub async fn run(runtime: Runtime, config: Config) {
         }
     };
 
-    // if "ros_localhost_only" is set, configure CycloneDDS to use only localhost interface
-    if config.ros_localhost_only {
-        if ros_distro_is_less_than("iron") {
+    // Dynamic Discovery is changed after iron. Need to check the ROS 2 version.
+    // https://docs.ros.org/en/rolling/Tutorials/Advanced/Improved-Dynamic-Discovery.html
+    if ros_distro_is_less_than("iron") {
+        // if "ros_localhost_only" is set, configure CycloneDDS to use only localhost interface
+        if config.ros_localhost_only {
             env::set_var(
                 "CYCLONEDDS_URI",
                 format!(
                     "{}{}",
-                    CYCLONEDDS_CONFIG_LOCALHOST_ONLY_BEFORE_HUMBLE,
+                    CYCLONEDDS_CONFIG_LOCALHOST_ONLY,
                     env::var("CYCLONEDDS_URI").unwrap_or_default()
                 ),
             );
+        }
+    } else {
+        let (ros_automatic_discovery_range, ros_static_peers) = if config.ros_localhost_only {
+            // If ROS_LOCALHOST_ONLY is set, need to transform into new environmental variables
```

Comment:
Here we should log the same warning than `rcl` in such situation:
```log
[WARN] [1732291697.959820404] [rcl]: ROS_LOCALHOST_ONLY is deprecated but still honored if it is enabled. Use ROS_AUTOMATIC_DISCOVERY_RANGE and ROS_STATIC_PEERS instead.
[WARN] [1732291697.959843184] [rcl]: 'localhost_only' is enabled, 'automatic_discovery_range' and 'static_peers' will be ignored.
```

Reply:
Make sense